internal/persistence/ent/client/sip.go

```diff
@@ -112,16 +132,35 @@ func (c *client) UpdateSIP(
 		q.SetFailedKey(up.FailedKey)
 	}
 
+	// If the UploaderID is set, find the user and set it on the query.
+	var userUUID *uuid.UUID
+	if up.UploaderID != nil {
+		u, err := c.ent.User.Query().Where(user.UUID(*up.UploaderID)).Only(ctx)
+		if err != nil {
+			return nil, rollback(tx, newDBError(errors.New("update SIP: uploader not found")))
+		}
+		q.SetUser(u)
+		userUUID = &u.UUID
+	}
+
```

Comment:
I'd not consider the uploader an updatable field, I'd just ignore it in the SIP update.